src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,178 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the Duke User Interface which deals with interactions with the user
+ */
+public class Ui {
+    public Ui() { }
+
+    /**
+     * Processes DukeException errors and displays them to the user.
+     *
+     * @param e The DukeException thrown
+     */
+    public void showLoadingError(DukeException e) {
+        System.out.println(e.getMessage());
+    }
+
+    public void greeting() {
+        System.out.println("Hello! I'm Duke\n" + "What can I do for you?");
+    }
+
+    /**
+     * Simulates an exit event when the user inputs the bye command.
+     *
+     * @return Returns "Bye" if the user inputs the bye command.
+     */
+    public String goodbye(boolean isRunningOnGui) {
+        if (isRunningOnGui) {
+            return "Bye! Hope to see you again soon!";
+        } else {
+            System.out.println("Bye! Hope to see you again soon!");
+            return "Bye";
+        }
+    }
+
+    /**
+     * Displays the user's current task list.
+     *
+     * @param tasks The current TaskList of the User.
+     */
+    public String list(TaskList tasks, boolean isRunningOnGui) {
+        if (isRunningOnGui) {
+            if (tasks.size() == 0) {
+                String noTasks = "There are no tasks in your list yet.";
+                return noTasks;
+            } else {
+                String response = "Here are the tasks in your list:";
+                for (int i = 0; i < tasks.getTasks().size(); i++) {
+                    response += "\n" + Integer.toString((i + 1)) + ". " + tasks.getTasks().get(i).recordString();
+                }
+                return response;
+            }
+        } else {
+            if (tasks.size() == 0) {
+                System.out.println("There are no tasks in your list yet.");
+                return "";
+            } else {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < tasks.getTasks().size(); i++) {
+                    System.out.println((i + 1) + ". " + tasks.getTasks().get(i).recordString());
+                }
+                return "";
+            }
+        }
+    }
+
+    /**
+     * Notifies the user that the specified task has been marked done.
+     *
+     * @param t The specified task that was marked done.
+     */
+    public String done(Task t, boolean isRunningOnGui) {
+        if (isRunningOnGui) {
+            String response = "Nice! I've marked this task as done:";
+            response += "\n" + t.recordString();
+            return response;
+        } else {
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(t.recordString());
+            return "";
+        }
+    }
+
+    /**
+     * Notifies the user that the specified task has been deleted.
+     * Notifies the user how many tasks are left in the current TaskList.
```

Comment:
Perhaps we can combine the comment into one line to make the comment more succinct?